atc/db/worker_resource_cache.go

```diff
@@ -71,14 +88,17 @@ func (workerResourceCache WorkerResourceCache) FindOrCreate(tx Tx, sourceWorkerB
 }
 
 // Find looks for a worker resource cache by resource cache id and worker name.
+// If returned worker_resource_cache's worker_base_resource_type_id is 0, that
+// means the source base resource type has been invalidated, thus the cache itself
+// can still be used, but it should no longer be cached again on other workers.
```

Comment:
> it should no longer be cached again on other workers

Just to confirm this is covered in the added line
```
workers, err := defaultTeam.FindWorkersForResourceCache(resourceCache.ID())
				Expect(err).ToNot(HaveOccurred())
				Expect(len(workers)).To(Equal(0))
```
in volume_test.go above

Reply:
Yes, as well as https://github.com/concourse/concourse/pull/8100/files#diff-a9320a1c83342789caf5df8171fc55be779a1892df352ad696aab0ff91e630d7R333